src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1529,6 +1529,100 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                 nuGetProjectContext, primarySources, secondarySources, token);
         }
 
+        // Preview and return ResolvedActions for many NuGetProjects.
+        public async Task<IEnumerable<ResolvedAction>> PreviewProjectsInstallPackageAsync(
+            IEnumerable<NuGetProject> nuGetProjects,
+            PackageIdentity packageIdentity,
+            ResolutionContext resolutionContext,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<SourceRepository> activeSources,
+            CancellationToken token)
+        {
+            if (nuGetProjects == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProjects));
+            }
+
+            if (packageIdentity == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            if (resolutionContext == null)
+            {
+                throw new ArgumentNullException(nameof(resolutionContext));
+            }
+
+            if (nuGetProjectContext == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProjectContext));
+            }
+
+            if (activeSources == null)
+            {
+                throw new ArgumentNullException(nameof(activeSources));
+            }
+
+            if (!activeSources.Any())
+            {
+                throw new ArgumentException("At least 1 item expected for " + nameof(activeSources));
+            }
+
+            if (packageIdentity.Version == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            var results = new List<ResolvedAction>();
+
+            // BuildIntegratedNuGetProject type projects are now supports parallel preview action for faster performance.
+            var buildIntegratedProjectsToUpdate = new List<BuildIntegratedNuGetProject>();
+            // Currently packages.config projects are not supported are supports parallel previews.
+            // Here is follow up issue to address it https://github.com/NuGet/Home/issues/9906
+            var otherTargetProjectsToUpdate = new List<NuGetProject>();
+
+            foreach (var proj in nuGetProjects)
+            {
+                if (proj is BuildIntegratedNuGetProject buildIntegratedNuGetProject)
+                {
+                    buildIntegratedProjectsToUpdate.Add(buildIntegratedNuGetProject);
+                }
+                else
+                {
+                    otherTargetProjectsToUpdate.Add(proj);
+                }
+            }
+
+            if (buildIntegratedProjectsToUpdate.Any())
```

Comment:
I would have to step into `.Any` to verify it, but I believe that it should be optimized to be equivalent to `.Count > 0`...

Reply:
@srdjanjovcic here is decompiled code for `Any()`.
![image](https://user-images.githubusercontent.com/8766776/91903696-858b5680-ec58-11ea-9ef3-9d533483f4e4.png)

Reply:
Which one is it extracted from? Dot net source browser says there are special cases for ICollection<>:

https://source.dot.net/#System.Linq/System/Linq/AnyAll.cs,8788153112b7ffd0

Reply:
typecasting (`if (value is ICollection)`) has a cost which is simply not needed if you use `.Count > 1` instead. I'm not saying it's a huge perf gain, but it doesn't require any kind of refactoring, and therefore doesn't meet the criteria for the famous "premature optimization is the root of all evil" justifaction.

Reply:
FWIW, I just ran a little benchmark with BenchmarkDotNet comparing `return x.Any()` vs `return x.Count > 0`.

```text
|  Method |           Job |       Runtime |       Mean |     Error |    StdDev |     Median |  Ratio | RatioSD |
|-------- |-------------- |-------------- |-----------:|----------:|----------:|-----------:|-------:|--------:|
| LinqAny |    .NET 4.7.2 |    .NET 4.7.2 | 21.9659 ns | 0.0822 ns | 0.0769 ns | 21.9814 ns |      ? |       ? |
|   Count |    .NET 4.7.2 |    .NET 4.7.2 |  0.0000 ns | 0.0000 ns | 0.0000 ns |  0.0000 ns |      ? |       ? |
|         |               |               |            |           |           |            |        |         |
| LinqAny | .NET Core 2.1 | .NET Core 2.1 | 20.4153 ns | 0.0645 ns | 0.0603 ns | 20.3939 ns |      ? |       ? |
|   Count | .NET Core 2.1 | .NET Core 2.1 |  0.0102 ns | 0.0169 ns | 0.0141 ns |  0.0026 ns |      ? |       ? |
|         |               |               |            |           |           |            |        |         |
| LinqAny | .NET Core 3.1 | .NET Core 3.1 | 16.5302 ns | 0.3592 ns | 0.7418 ns | 16.5427 ns | 401.23 |  114.15 |
|   Count | .NET Core 3.1 | .NET Core 3.1 |  0.0454 ns | 0.0179 ns | 0.0150 ns |  0.0455 ns |   1.00 |    0.00 |
|         |               |               |            |           |           |            |        |         |
| LinqAny | .NET Core 5.0 | .NET Core 5.0 |  4.0125 ns | 0.0253 ns | 0.0237 ns |  4.0006 ns |      ? |       ? |
|   Count | .NET Core 5.0 | .NET Core 5.0 |  0.0000 ns | 0.0000 ns | 0.0000 ns |  0.0000 ns |      ? |       ? |

// * Warnings *
BaselineCustomAnalyzer
  Summary -> A question mark '?' symbol indicates that it was not possible to compute the (Ratio, RatioSD) column(s) because the baseline value is too close to zero.
ZeroMeasurement
  Program.Count: .NET 4.7.2    -> The method duration is indistinguishable from the empty method duration
  Program.Count: .NET Core 2.1 -> The method duration is indistinguishable from the empty method duration
  Program.Count: .NET Core 5.0 -> The method duration is indistinguishable from the empty method duration
```

`.Any()` may only take nanoseconds, but you can still see how much slower it is than `.Count > 0`. Multiple these nano-second time wastage by the hundreds or thousands of times this anti-pattern is used thoughout a codebase, and you can end up with multi-second delays. Since `.Any()` takes much, much less than 1ms, a performance trace is unlikely to capture it in a stack trace, so might not even tell you about this poor performance practice.

Reply:
Ok. Taken the suggestion.

Reply:
I would do search'n'replace on all `.Any()` calls throughout codebase, then just undo those which do not work (on `IEnumerable<>`, for example). 🙂

Reply:
...as a separate PR, off course.